packages/lisk-p2p/src/p2p.ts

```diff
@@ -92,6 +101,11 @@ export const DEFAULT_DISCOVERY_INTERVAL = 30000;
 
 const BASE_10_RADIX = 10;
 
+const selectRandomPeerSample = (
```

Comment:
This function is now in two places; here and in peerPool. We should try to reuse. Maybe we can export the one from the PeerPool and use that one?

Reply:
Oh actually there is a small difference, this one is using `P2PDiscoveredPeer` and the other one is using `Peer` objects. Or maybe, I can make it a general function that only takes `P2PDiscoveredPeer` and in PeerPool I pass the list of peerinfos